View/script.js

```diff
@@ -100,106 +102,161 @@ const updateDivs = (gameObjArr, divs) => {
-    });
-    game.spaceships.forEach((ship, i) => {
-      ship.update();
-      spaceshipDivs[i].style.left = `${ship.x}px`;
-    });
-    game.amongs.forEach((among, i) => {
-      among.update();
-      amongDivs[i].style.left = `${among.x}px`;
-    });
-    game.stones.forEach((stone, i) => {
-      stone.update();
-      stoneDivs[i].style.left = `${stone.x}px`;
-    });
-    game.bananas.forEach((banana, i) => {
-      banana.update();
-      bananaDivs[i].style.left = `${banana.x}px`;
-    });
-    game.trashs.forEach((trash, i) => {
-      trash.update();
-      trashDivs[i].style.left = `${trash.x}px`;
+    let gameObjsAndDivs = [
+      { obj: game.cars, div: carDivs },
+      { obj: game.spaceships, div: spaceshipDivs },
+      { obj: game.amongs, div: amongDivs },
+      { obj: game.stones, div: stoneDivs },
+      { obj: game.bananas, div: bananaDivs },
+      { obj: game.trashs, div: trashDivs },
+    ];
+
+    // Updating obstacles div's position
+    gameObjsAndDivs.forEach((pair) => {
+      updateDivs(pair.obj, pair.div);
     });
+
     Object.keys(game.eventHandler.eventList).forEach((key) => {
       game.eventHandler.trigger(key, game.player);
       game.eventHandler.off(key);
     });
+
     playerDiv.style.left = `${game.player.x}px`;
     playerDiv.style.top = `${game.player.y}px`;
 
+    // Checking if one of the obstacles has collisoned into our player
     [
-      game.cars,
-      game.spaceships,
-      game.amongs,
-      game.stones,
-      game.bananas,
-      game.trashs,
-    ].forEach((list) => {
-      list.forEach((obj, index) => {
-        if (game.player.collision(obj)) {
-          if (obj.isGood) {
-            game.player.update();
-          } else {
-            game.lives--;
-            initGame();
-          }
-        }
-      });
-    });
->>>>>>> 34e6b94... Fixed bug when after re-creating game collision didnt work
-
-  allObjs.forEach((list) => {
-    list.forEach((obj, index) => {
+      ...game.cars,
+      ...game.spaceships,
+      ...game.amongs,
+      ...game.stones,
+      ...game.bananas,
+      ...game.trashs,
+    ].forEach((obj) => {
       if (game.player.collision(obj)) {
         if (obj.isGood) {
-          game.player.update();
+          // console.log(obj.dir);
+          game.player.update(obj.dir);
         } else {
           game.lives--;
           initGame();
         }
       }
     });
-  });
 
-  if (
-    game.screenY * 0.1 <= game.player.y <= game.screenY * 0.3 &&
-    game.player.collide
-  )
-    game.isPlaying = false;
+    // Checking if our player has landed on one of the docks and if he did
+    // changing the state of the game according to the current state
+    game.docks.forEach((dock, index) => {
+      if (game.player.collision(dock) && dock.isAvailable) {
+        dock.isAvailable = false;
+        rocketDivs[index].style.visibility = "visible";
+        switch (`${game.player.name}-${game.player.model}`) {
+          case "Ofeq-5":
```

Comment:
Again, most cases here are almost identical. Shorten the lines, and make it all more readable.